### ISSUE TITLE
[Android] OOM telemetry api

### ISSUE THREAD

**Issue by Larnoo** [mapbox/mapbox-gl-native#9262]
Mapbox SDK： v 5.0.0

Logs are as follow.
```
java.lang.OutOfMemoryError: Failed to allocate a 25459360 byte allocation with 8388608 free bytes and 10MB until OOM
	at java.lang.AbstractStringBuilder.enlargeBuffer(AbstractStringBuilder.java:95)
	at java.lang.AbstractStringBuilder.append0(AbstractStringBuilder.java:133)
	at java.lang.StringBuilder.append(StringBuilder.java:124)
	at org.json.JSONStringer.string(JSONStringer.java:344)
	at org.json.JSONStringer.key(JSONStringer.java:375)
	at org.json.JSONObject.writeTo(JSONObject.java:720)
	at org.json.JSONStringer.value(JSONStringer.java:237)
	at org.json.JSONArray.writeTo(JSONArray.java:613)
	at org.json.JSONArray.toString(JSONArray.java:585)
	at com.mapbox.services.android.telemetry.c.b.sendEvents(TelemetryClient.java:1202)
	at com.mapbox.services.android.telemetry.b.a(MapboxTelemetry.java:515)
	at com.mapbox.services.android.telemetry.b.a(MapboxTelemetry.java:402)
	at com.mapbox.services.android.telemetry.TelemetryLocationReceiver.onReceive(TelemetryLocationReceiver.java:1444)
	at android.support.v4.content.LocalBroadcastManager.executePendingBroadcasts(LocalBroadcastManager.java:297)
	at android.support.v4.content.LocalBroadcastManager.access$000(LocalBroadcastManager.java:46)
	at android.support.v4.content.LocalBroadcastManager$1.handleMessage(LocalBroadcastManager.java:116)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:150)
	at android.app.ActivityThread.main(ActivityThread.java:5621)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:794)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:684)
```

why the code need request 24M memory?
 
Can you tell me the `telemetry` package  functions?

**Comment by Larnoo** [comment]
```
java.lang.OutOfMemoryError: Failed to allocate a 19593630 byte allocation with 2097152 free bytes and 10MB until OOM
	at java.lang.StringFactory.newStringFromChars(StringFactory.java)
	at java.lang.AbstractStringBuilder.toString(AbstractStringBuilder.java:629)
	at java.lang.StringBuilder.toString(StringBuilder.java:663)
	at org.json.JSONStringer.toString(JSONStringer.java:430)
	at org.json.JSONArray.toString(JSONArray.java:586)
	at com.mapbox.services.android.telemetry.http.TelemetryClient.sendEventsWrapped(TelemetryClient.java:202)
	at com.mapbox.services.android.telemetry.http.TelemetryClient.sendEvents(TelemetryClient.java:117)
	at com.mapbox.services.android.telemetry.MapboxTelemetry.flushEventsQueueImmediately(MapboxTelemetry.java:515)
	at com.mapbox.services.android.telemetry.MapboxTelemetry.putEventOnQueue(MapboxTelemetry.java:402)
	at com.mapbox.services.android.telemetry.MapboxTelemetry.addLocationEvent(MapboxTelemetry.java:444)
	at com.mapbox.services.android.telemetry.TelemetryLocationReceiver.onReceive(TelemetryLocationReceiver.java:32)
	at android.support.v4.content.LocalBroadcastManager.executePendingBroadcasts(LocalBroadcastManager.java:297)
	at android.support.v4.content.LocalBroadcastManager.access$000(LocalBroadcastManager.java:46)
	at android.support.v4.content.LocalBroadcastManager$1.handleMessage(LocalBroadcastManager.java:116)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:150)
	at android.app.ActivityThread.main(ActivityThread.java:5546)
	at java.lang.reflect.Method.invoke(Method.java)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:792)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:682)
```

**Comment by tobrun** [comment]
Duplicate of https://github.com/mapbox/mapbox-gl-native/issues/9131, resolved with 5.1.0, feel free to retest with v5.1.0-beta.4. Thank you for reaching out and reporting your problem!
